### ISSUE TITLE
Not Real Issue but Question

### ISSUE THREAD

**Issue by mjs513** [danzimmerman/barebones_MAVLink#1]
Hi @danzimmerman 
I came across this while searching for handling param_set command from the GCS to a Teensy 3.5.  Have to tell you, nice job on getting it working.  Will help a great deal.

Been working on mine for a couple of months now.  I am pretty much able to send message to the GCS without any issue but having 1 last challenge maybe you can help me with.  Have you had any luck with setting parameter values from the GCS to the Teensy?  I am seeing that the message is sent but not seeing the message come through on the Teensy (since I am debugging and have the Message ids printed :).  

Any suggestions would be appreciated.

Thanks
Mike

**Comment by mjs513** [comment]
After reading your blog again I resolved the problem and am now able to get the message.  I had to change comp id to 1 and all is well with the world.  With the help of you sketch I can updated my implementation to finish command long handling,  set params and clean up some other code.

Really want to thank you again for your efforts on this.  Wish I found it a lot sooner 

Mike

**Comment by danzimmerman** [comment]
Glad you got it working and this was helpful! Are you using the Blue Robotics QGC build or the official, general-purpose one?

**Comment by mjs513** [comment]
Sorry, just saw your message, was working on the code.
 Is the Blue robotics QGC build any better?

I am working with generic c_version1 library, probably should switch over to v2 but started with v1 and have gps and ahrs data flowing nicely to APM planner.

Still have problems with just doing a single set parameter change.  The only way to do it seems to be clicking on the write button from the tuning window.  If I just click on set nothing comes across.  Wouldn't be so bad but I don't see the updated value for a single parameter.  If I have more that one that I upload I always just see the last one with the original value.  My problem is just don't see how's its done.  The protocol description is rather useless for an actual implementation.  Struggling with the proper command ack format.  Right now just sending an ack with an accepted.  Know I am missing something just don't know what.  This is actually the last piece of the puzzle.

Mavlink is like pulling teeth.

**Comment by mjs513** [comment]
Hi Dan
I just loaded your example onto a T3.5 just to test and play around with the messages.

1.  On the APM plannerI had to update the capabilities message to include the mavlink2 protocal enum.  But I am still getting a strange error that there is a baud rate mismatch between the T3.5 and the ground station.  Strange, both are set at 115200. 
2. On QGCS I am able to connect no problem but it does not seem to want to arm.  Gives me a can not arm message.  Strange.
Just thought you would like to know.

**Comment by danzimmerman** [comment]
Thanks for the info. 

I just tried this with both Blue Robotics QGC version 3.2.4 and the newly downloaded official version QGC 3.3.2 on a Teensy 3.2 and didn't have a problem. Arm/disarm seems to work okay there. I also tried adding MAV_PROTOCOL_CAPABILITY_MAVLINK2 to the capabilities message:

![image](https://user-images.githubusercontent.com/4365714/43089327-563d3416-8e72-11e8-9a70-bf1af58aa4f6.png)

to see if that was the issue. Arm/disarm still seems to still work in both versions of QGC that I have. 

![image](https://user-images.githubusercontent.com/4365714/43089475-b92fcc28-8e72-11e8-8465-200b6f8f3f19.png)

When you say you're using the  `c_version1` library, did you download my sketch by itself and use your local copy of the library, or did you clone the whole repository and compile including the submodule `mavlink_c_v2` library linked from the repository? 

I don't know what to say about your baud rate error. I downloaded APM planner 2.0.24 (on Win 10) and tried it and it connects to the Teensy's COM port okay. It doesn't seem to actually --work-- though, but I don't know APM planner and how to get it running.

**Comment by mjs513** [comment]
Thanks for getting back to me.  Guess I have a little more trouble shooting to do.

For your sketch I was using the library that you used.  For both the QGC and APM planner.  For apm planner I was using 2.0.26r2.  Was having a issue with UDP, when I was using mavesp8266.  I switched to 3dr clones radios.

I'll give it some more testing and download QGC again just in case.  Its probably me, I will keep you posted.  Eventually I will post my sloppy code.  Have to eventually clean it up, probably go your way.  A lot cleaner. 

Thanks for your help.
Mike

**Comment by mjs513** [comment]
Ok,  still couldn't get it to connect to standard QGC.  However, I downloaded the BlueROV QGC.  Connected and armed without a problem using your sketch.

I did get it to connect to APM planner by changing the baud rate to 57600 but nothing came through.  So for the heck of it I changed the library to mavlink version 1, yes I know.  But guess what.  It connected at 57k and 115k no problem.  Also, was receiving data across the link.  Only thing couldn't do was arm the system which also requires changing the mode.

Will keep working at it, to see what I can come up with.
